packages/next/shared/lib/match-remote-pattern.ts

```diff
@@ -0,0 +1,68 @@
+import type { RemotePattern } from './image-config'
+
+export function matchRemotePattern(pattern: RemotePattern, url: URL): boolean {
+  if (pattern.protocol !== undefined) {
+    const actualProto = url.protocol.slice(0, -1)
+    if (pattern.protocol !== actualProto) {
+      return false
+    }
+  }
+  if (pattern.port !== undefined) {
+    if (pattern.port !== url.port) {
+      return false
+    }
+  }
+  if (pattern.pathname !== undefined) {
+    const patternParts = pattern.pathname.split('/')
+    const actualParts = url.pathname.split('/')
+    const len = Math.max(patternParts.length, actualParts.length)
+    for (let i = 0; i < len; i++) {
+      if (patternParts[i] === '**' && actualParts[i] !== undefined) {
```

Comment:
Does this mean we can't ensure the path ends with a specific value e.g. `/hello/world/image` with `{ pathname: '/**/image' }`?

If so would be good to validate `**` is the end value and nothing is added afterwards/and mention this in the documentation

Reply:
Done! https://github.com/vercel/next.js/pull/36245#discussion_r865493579